### PR TITLE
extend new barrel lint rule to require certain barrel usage internally

### PR DESCRIPTION
**Pull request by MichaelBelousov**
- add "required-barrel-modules" option to "no-internal-barrel-imports" eslint rule
- fixed incorrect importing of files from `tile/` without the `tile/internal` barrel
- added map/MapTiledGraphicsProvider to the `tile/internal` barrel file

### CODE REVIEW

File: core/frontend/src/Viewport.ts
Comment (MichaelBelousov) [inline core/frontend/src/Viewport.ts:52]: @pmconne is there a reason this file wasn't in the barrel? I added it to the barrel, but I'm not sure if I added it in the right place considering it could introduce a cyclic dependency

Comment (pmconne) [reply core/frontend/src/Viewport.ts:52]: Nope, just an oversight on somebody's part. Location in internal.ts looks ok.

**Review by pmconne — APPROVED** [review]

File: common/api/ecschema-metadata.api.md
Comment (pmconne) [inline common/api/ecschema-metadata.api.md:2069]: You may need to revert these changes. extract-api is super-flaky about the ecschema-metadata package for some reason.

Comment (MichaelBelousov) [reply common/api/ecschema-metadata.api.md]: last time this happened caleb recommended `rush install && rush clean && rush build && rush extract-api`, I just pushed the result of that and believe it should be fixed

**Comment by MichaelBelousov** [comment]
@mergifyio update

**Comment by MichaelBelousov** [comment]
@mergifyio update

**Comment by pmconne** [comment]
@calebmshafer how does @MichaelBelousov get a build queued when branch is already up to date?

**Comment by MichaelBelousov** [comment]
> @calebmshafer how does @MichaelBelousov get a build queued when branch is already up to date?

I manually triggered one... I still have an extract-api issue after a clean+build reverted the bad ecschema-metadata api change

**Comment by pmconne** [comment]
> I manually triggered one... 

How, exactly? With "run new" in ADO? I don't think that will suffice.

**Comment by MichaelBelousov** [comment]
> How, exactly? With "run new" in ADO? I don't think that will suffice.

Yes, with `run new`

**Comment by pmconne** [comment]
You can either wait for Caleb, or wait for someone else to merge to master, or push some trivial change to a comment somewhere.

**Comment by calebmshafer** [comment]
@MichaelBelousov, @pmconne is correct just doing a "run new" will not work because it won't be associated with the PR but only the branch. 

It looks like in this case ADOps just missed the webhook of the PR. The only way I know how to fix that is to either push another change or close and re-open the PR.

**Comment by MichaelBelousov** [comment]
I didn't think it would work for that reason, exactly what I did was:
- see if mergify-io update would rerun the check
- see if `rerun failed jobs` would rerun the check
- see if `run new` would rerun the check.
now that I know it shouldn't work I won't try that again, but it did seem to trigger a build

**Comment by pmconne** [comment]
I vote to force merge this. iModelHubClient integration tests failed, and imodeljs-backend tests failed on Windows only.
@calebmshafer ok with you? FYI PR is awaiting your approval.

**Review by DanEastBentley — APPROVED** [review]

**Comment by pmconne** [comment]
@Mergifyio update

**Comment by pmconne** [comment]
@calebmshafer @paulius-valiunas please approve PR.

**Pull request merged**
